### PR TITLE
Added ruff linting to the server

### LINKED ISSUE #19: Add linting to the server
Adding linting to projects is a great way to catch errors and ensure the code quality stays high. 
[Ruff](https://beta.ruff.rs/docs/) is a linter for python that is written in [Rust](https://www.rust-lang.org/) and extremely fast.
Adding it to the server (as a [pdm script](https://pdm.fming.dev/latest/usage/scripts/) initially) would be  useful both during development (where tools like [Ale](https://github.com/dense-analysis/ale#asynchronous-lint-engine---) or a [VSCode extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) surface linting errors as you type), and also in the CI pipeline that we'll be creating shortly.

### PR DESCRIPTION
**Pull request by andGuo**
Resolved #19. Added ruff with pdm script and fixed max line length errors. Did not fix ruff unused import errors.

Run with command:
```
pdm run linter
```

Probably should move ruff to a .pre-commit-config.yaml in the future CI pipeline.

### CODE REVIEW

**Comment by andGuo** [comment]
Also, if only ruff is going to be used, Black can be removed from the server dev dependencies.

**Review by szeckirjr — APPROVED** [review]
LGTM 🎉

**Comment by sleepycat** [comment]
Thanks for this @andGuo! Can you rebase this and we'll get it merged!

File: server/pyproject.toml [hunk 1]
```diff
@@ -60,5 +60,8 @@ dev = {composite = ["db", "wait", "migrate", "start"], env = {DB_HOST = "localho
 test = {cmd = "coverage run manage.py test -v 2", env = {DB_HOST = "localhost"}}
 citest = {cmd = "python manage.py test -v 2 --keepdb"}
 post_test = {cmd = "coverage report --skip-empty --omit=tests/*,*tests*"} #choose additional files to skip here with "," and no spaces)
+linter = {cmd = "ruff check ."}
 
+[tool.ruff]
+line-length = 119 # Alternatives PEP-8: 79, Black/Ruff: 88
```
Comment (andGuo) [inline server/pyproject.toml:66]: 119 is the recommended length for Django. It is also the max char width for GitHub code review without needing to horizontal scroll. The alternatives will split many lines of Django in the server.

**Pull request merged**